core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -1009,8 +1009,8 @@ wasi_sock_accept(wasm_exec_env_t exec_env, wasi_fd_t fd, wasi_fd_t *fd_new)
 }
 
 static wasi_errno_t
-wasi_sock_addr_local(wasm_exec_env_t exec_env, wasi_fd_t fd, uint8 *buf,
-                     wasi_size_t buf_len)
+wasi_sock_addr_local(wasm_exec_env_t exec_env, wasi_fd_t fd,
+                     __wasi_addr_t *addr)
```

Comment:
Should check whether addr is out of the range in linear memory, as the signature is register as "(i*)i", addr is pointer argument, runtime only checks its first byte isn't out of range before passing it into native API:
```
if (!validate_native_addr(addr, sizeof(__wasi_addr_t))
    return error;
```

Reply:
that's good to know, thanks for pointing that out.